site.ts

```diff
@@ -220,22 +221,38 @@ export function inject({ config, posthog }) {
         })
     }
 
-    posthog?.getActiveMatchingSurveys((surveys) => {
-        surveys.forEach((survey) => {
-            const surveyName = survey.name.replace(/\s/g , "-")
-            if (!localStorage.getItem(`seenSurvey_${surveyName}_${survey.id}`)) {
-                const shadow = createShadow(style(surveyName, survey?.appearance))
-                const surveyPopup = createSurveyPopup(survey, surveyName)
-                addListeners(surveyPopup, surveyName, survey.id, survey.name)
-                shadow.appendChild(surveyPopup)
-                window.dispatchEvent(new Event('PHSurveyShown'))
-                const sessionRecordingUrl = getSessionRecordingUrl()
-                posthog.capture('survey shown', {
-                    $survey_name: survey.name,
-                    $survey_id: survey.id,
-                    sessionRecordingUrl: sessionRecordingUrl,
-                })
+    const callSurveys = (posthog) => {
+        posthog?.getActiveMatchingSurveys((surveys) => {
+            surveys.forEach((survey) => {
```

Comment:
this now leads to this weird edge case:

1. I'm shown surveyA, on page1
2. I navigate to page2, and then start filling in surveyA.
3. The interval runs, and adds a new survey on top of the one I already have. (judging by the code, doesn't seem like we remove the current one, which is good).

I'd recommend adding handling here for showing exactly one survey at a time - i.e. as long as a survey is shown, no new surveys are created.

Same for when more than one survey matches, instead of creating all of them, create just one and skip the rest.

With this, you don't need the duplicate handling you've added below on L227 either.

Reply:
That makes sense and I think is a good workaround for now. Eventually I could see users wanting to display different survey types at once (though they probably still shouldn't) but we'll cross that bridge when we get to it